Extensions/Lighting/tests/lightruntimeobject.spec.js

```diff
@@ -0,0 +1,355 @@
+      100,
+      300,
+      100,
+      300,
+      100.0199966430664,
+      300,
+      249.9875030517578,
+      300,
+      250,
+      299.9750061035156,
+      250,
+      250.00999450683594,
+      250,
+      250,
+      250,
+      250,
+      250.00999450683594,
+      250,
+      299.9750061035156,
+      250,
+      300,
+      249.9875030517578,
+      300,
+      100.0199966430664,
+      300,
+      100,
+      300,
+      100,
+      299.9800109863281,
+    ];
+    const indexData = [
+      0,
+      1,
+      2,
+      0,
+      2,
+      3,
+      0,
+      3,
+      4,
+      0,
+      4,
+      5,
+      0,
+      5,
+      6,
+      0,
+      6,
+      7,
+      0,
+      7,
+      8,
+      0,
+      8,
+      9,
+      0,
+      9,
+      10,
+      0,
+      10,
+      11,
+      0,
+      11,
+      12,
+      0,
+      12,
+      13,
+      0,
+      13,
+      14,
+      0,
+      14,
+      15,
+      0,
+      15,
+      16,
+      0,
+      16,
+      17,
+      0,
+      17,
+      18,
+      0,
+      18,
+      1,
+    ];
+
+    vertexData.forEach((val, index) => {
+      expect(vertexBuffer[index]).to.be(val);
+    });
+    indexData.forEach((val, index) => {
+      expect(indexBuffer[index]).to.be(val);
+    });
+  });
+
+  it('Vertex and index buffers after obstacle is moved.', function () {
+    obstacle.setPosition(150, 250);
+    runtimeScene.renderAndStep();
+    light.update();
+
+    const vertexBuffer = light._renderer._vertexBuffer;
+    const indexBuffer = light._renderer._indexBuffer;
+    const vertexData = [
+      200,
+      200,
+      100,
+      100.0199966430664,
+      100,
+      100,
+      100.0199966430664,
+      100,
+      299.9800109863281,
+      100,
+      300,
+      100,
+      300,
+      100.0199966430664,
+      300,
+      299.9800109863281,
+      300,
+      300,
+      299.9800109863281,
+      300,
+      200.00999450683594,
+      300,
+      200,
+      250,
+      199.9949951171875,
+      250,
+      175.00625610351562,
+      250,
+      175,
+      250,
+      174.99374389648438,
+      250,
+      150.00999450683594,
+      250,
+      150,
+      250,
+      100,
+      299.9800109863281,
+    ];
+    const indexData = [
+      0,
+      1,
+      2,
+      0,
+      2,
+      3,
+      0,
+      3,
+      4,
+      0,
+      4,
+      5,
+      0,
+      5,
+      6,
+      0,
+      6,
+      7,
+      0,
+      7,
+      8,
+      0,
+      8,
+      9,
+      0,
+      9,
+      10,
+      0,
+      10,
+      11,
+      0,
+      11,
+      12,
+      0,
+      12,
+      13,
+      0,
+      13,
+      14,
+      0,
+      14,
+      15,
+      0,
+      15,
+      16,
+      0,
+      16,
+      17,
+      0,
+      17,
+      18,
+      0,
+      18,
+      1,
+    ];
+
+    vertexData.forEach((val, index) => {
+      expect(vertexBuffer[index]).to.be(val);
+    });
+    indexData.forEach((val, index) => {
+      expect(indexBuffer[index]).to.be(val);
```

Comment:
ubernit: if vertexData and vertexBuffer have the same "things", they should have the same name. Instead of changing between "data" and "buffer" (that are synonym but create confusion about "what's different so that they choose another name?"), add `expected` to the name of the arrays that are used for the test:
```suggestion
    expectedVertexBuffer.forEach((val, index) => {
      expect(vertexBuffer[index]).to.be(val);
    });
    expectedIndexBuffer.forEach((val, index) => {
      expect(indexBuffer[index]).to.be(val);
```